### PR TITLE
Upgrade GPTQ

### PR DESCRIPTION
**Pull request by haihabi**
1. Enable different max change lsb for each bit-width.
2. Enable bias correction with GPTQ to improve initial training point.
3. Add Mixed-precision API config override option for bug.

### CODE REVIEW

File: model_compression_toolkit/common/gptq/gptq_config.py [hunk 1]
```diff
@@ -26,7 +28,8 @@ def __init__(self,
                  optimizer: Any,
                  loss: Callable = None,
                  log_function: Callable = None,
-                 train_bias: bool = True):
+                 train_bias: bool = True,
+                 lsb_change_per_bit_width=MAX_LSBS_CHANGE_MAP):
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/gptq/gptq_config.py:32]: Type-hints

File: model_compression_toolkit/common/mixed_precision/mixed_precision_quantization_config.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ def __init__(self,
                  weights_n_bits: List[int] = None,
                  compute_distance_fn: Callable = compute_mse,
                  distance_weighting_method: Callable = get_average_weights,
-                 num_of_images: int = 32):
-
+                 num_of_images: int = 32,
+                 configuration_overwrite: List[int] = None):
         """
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/mixed_precision/mixed_precision_quantization_config.py:33]: Are you sure you want to push this? It's for debug purposes

File: model_compression_toolkit/keras/back2framework/model_builder.py [hunk 3]
```diff
@@ -179,10 +181,13 @@ def model_builder(graph: common.Graph,
         fw_info: Framework information (e.g., mapping from layers to their attributes to quantize).
         This is needed when using MIXEDPRECISION or GPTQ mode for passing the kernel attributes to
         the QuanteWrapper we use in both of these cases.
+        gptq_config: GPRQ Configuration class.
```
Comment (reuvenperetz) [inline model_compression_toolkit/keras/back2framework/model_builder.py:184]: gptq_config: GPTQ Configuration class.

File: model_compression_toolkit/keras/quantizer/gradient_ptq/config_factory.py [hunk 4]
```diff
@@ -23,22 +23,21 @@
...
+                                     fw_info: FrameworkInfo,
+                                     gptq_config: GradientPTQConfig) -> QuantizeConfig:
     """
     Build a QuantizeConfig for a node according to its quantization configuration and
     a global NoOpQuantizeConfig object.
 
     Args:
         n: Node to build its QuantizeConfig.
         fw_info: Framework information (e.g., mapping from layers to their attributes to quantize).
+        gptq_config:
```
Comment (reuvenperetz) [inline model_compression_toolkit/keras/quantizer/gradient_ptq/config_factory.py:40]: gptq_config: GradientPTQConfig object for building QuantizeConfigs with max LSB change mapping.

File: model_compression_toolkit/keras/quantizer/gradient_ptq/weight_quantizer.py [hunk 5]
```diff
@@ -22,6 +22,14 @@
...
 from model_compression_toolkit.keras.quantizer.gradient_ptq.utils import symmetric_constrained_quantizer
 from model_compression_toolkit.common.constants import THRESHOLD
+from model_compression_toolkit.common.defaultdict import DefaultDict
+
+
+def get_kernel(weights_list):
+    for w in weights_list:
+        if "kernel" in w.name:
+            return w
+    raise Exception("Can't find kernel variable")
```
Comment (reuvenperetz) [inline model_compression_toolkit/keras/quantizer/gradient_ptq/weight_quantizer.py:32]: Typehints, comments, and please use the constant "kernel" from keras constants

Comment (reuvenperetz) [reply model_compression_toolkit/keras/quantizer/gradient_ptq/weight_quantizer.py:32]: Or maybe the kernel attribute should be taken from fw_info... What do you think?

File: model_compression_toolkit/common/mixed_precision/mixed_precision_quantization_config.py (hunk 2 above)
Comment (haihabi) [reply model_compression_toolkit/common/mixed_precision/mixed_precision_quantization_config.py:33]: yes

**Pull request merged**